test/common/formatter/substitution_format_string_test.cc

```diff
@@ -135,5 +135,38 @@ TEST_F(SubstitutionFormatStringUtilsTest, TestFromProtoConfigFormatterExtensionU
                             "Formatter not found: envoy.formatter.TestFormatterUnknown");
 }
 
+TEST_F(SubstitutionFormatStringUtilsTest, TestFromProtoConfigJsonWithExtension) {
+  TestCommandFactory factory;
+  Registry::InjectFactory<CommandParserFactory> command_register(factory);
+
+  const std::string yaml = R"EOF(
+  json_format:
+    text: "plain text %COMMAND_EXTENSION()%"
```

Comment:
Could you add a test to cover multiple formatters?

Reply:
you mean mixing internal and extension formatters or just multiple extensions?

Reply:
Not sure if you need to test expansion of more than one substitution string: a single working substitution proves that the non-extended formatter works, the rest of the checks is adding to the noise (especially if there's a regression in the formatter that leads to multiple test failures).

Reply:
> you mean mixing internal and extension formatters or just multiple extensions?

Multiple extensions -- multiple can be configured, but we don't seem to test that anywhere.

Reply:
I think multiple expansions are probably covered in #14512 (this just covers json format). However, invoking the extension formatter from other parts of the json dict is probs a good idea.

Reply:
> a single working substitution proves that the non-extended formatter works, the rest of the checks is adding to the noise (especially if there's a regression in the formatter that leads to multiple test failures).

Well, it's still good to exercise nested structures to see if all is good -- but as mentioned above I'll invoke the extension from other values in the dict too.

I can tested multiple extensions too, np.

Reply:
> I can test multiple extensions too, np.

Thanks, I think this is more of an intergration-level type check, but I don't see an existing test covering this case. If there's one that I missed, then adding one specific to json format is probably excessive.

Reply:
Actually you are right, we didn't have a test for multiple extensions so I think it's a good idea to piggy back on this change and add something. I'll push something in a bit. Thanks!

Reply:
Ok done